tests/periph_backup_ram/main.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Low-level test for backup RAM
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "periph/rtc.h"
+#include "pm_layered.h"
+#include "xtimer.h"
+
+#ifndef SLEEP_SEC
+#define SLEEP_SEC 1
+#endif
+
+int main(void)
+{
+    /* We keep a copy of counter in the .noinit section (normal RAM).
+     * In the deepest sleep normal RAM is not retained, so if it matches
+     * counter anyway after wakeup, we did not sleep properly.
+     */
+    static int counter_noinit __attribute__((section(".noinit")));
+    static int counter __attribute__((section(".backup.bss")));
+
+    if (counter == 0) {
+        puts("\nBackup RAM test\n");
+        printf("This test will increment the counter by 1, "
+               "then enter deep sleep for %ds\n", SLEEP_SEC);
+    } else if (counter_noinit == counter) {
+        puts("WARNING: non-backup memory retained - did we really enter deep sleep?");
+    }
+
```

Comment:
My personal taste would have place `#ifndef CPU_BACKUP_RAM_NOT_RETAINED` here so SAML21 platform doesn't display the counter before saying "aborting test". You may ignore this comment if you're fine this way.

Reply:
I think we should at least once access that memory to make sure it doesn't crash & burn when doing so.